lib/ts/recipe/emailverification/emailVerificationClaim.ts

```diff
@@ -0,0 +1,54 @@
+import { SessionClaimValidator, BooleanClaim } from "../session";
+import { RecipeInterface } from "./types";
+
+/**
+ * We include "Class" in the class name, because it makes it easier to import/use the right thing (the instance exported by the recipe) instead of this.
+ * */
+export class EmailVerificationClaimClass extends BooleanClaim {
+    constructor(getRecipeImpl: () => RecipeInterface) {
+        super({
+            id: "st-ev",
+            refresh: async (userContext) => {
+                await getRecipeImpl().isEmailVerified({
+                    userContext,
+                });
+            },
+        });
+
+        this.validators = {
+            ...this.validators,
+            isVerified: (refetchTimeOnFalseInSeconds = 10, updateContextOnInvalidClaim) => ({
+                id: this.id,
+                refresh: this.refresh,
+                shouldRefresh: (payload, userContext) => {
+                    const value = this.getValueFromPayload(payload, userContext);
+                    return (
+                        value === undefined ||
+                        (value === false &&
+                            this.getLastFetchedTime(payload, userContext)! <
+                                Date.now() - refetchTimeOnFalseInSeconds * 1000)
+                    );
+                },
+                validate: async (payload, userContext) => {
+                    const value = this.getValueFromPayload(payload, userContext);
+                    if (value !== true && updateContextOnInvalidClaim !== undefined) {
+                        await updateContextOnInvalidClaim(userContext);
+                    }
+                    return value === true
+                        ? { isValid: true }
+                        : {
+                              isValid: false,
+                              reason: { message: "wrong value", expectedValue: true, actualValue: value },
+                          };
+                },
+            }),
+        };
+    }
+
+    validators!: BooleanClaim["validators"] & {
+        isVerified: (
+            refetchTimeOnFalseInSeconds?: number,
+            updateContextOnInvalidClaim?: (userContext: any) => void | Promise<void>
+        ) => SessionClaimValidator;
```

Comment:
Instead of making this take a callback to update the context, why not just give it the updated context when it's being called?
```
isVerified: (refetchTimeOnFalseInSeconds?: number, userContext: any) => SessionClaimValidator
```

Reply:
this is called during `init`, meaning we don't have access to `userContext`, also, there may be different userContexts used in different validation calls.

Reply:
We can move the `updateContextOnInvalidClaim` arg into the constructor instead.